src/sinks/elasticsearch.rs

```diff
@@ -402,26 +402,20 @@ fn signed_request(method: &str, uri: &Uri) -> SignedRequest {
 
 /// Derive a region name from a hostname
 fn derive_region(uri: &Uri) -> Region {
+    // Standard AWS ES service domain names look like:
+    // DOMAIN-CODE.REGION.es.amazonaws.com
+    // DOMAIN-CODE.REGION.es.amazonaws.com.cn
+    // So, split the domain name on '.' and take the second part.
     uri.host()
-        // Strip the standard domain name top level
         .and_then(|hostname| {
-            if hostname.ends_with(".amazonaws.com") {
-                Some(&hostname[..hostname.len() - 14])
-            } else if hostname.ends_with(".amazonaws.com.cn") {
-                Some(&hostname[..hostname.len() - 17])
+            let parts: Vec<&str> = hostname.split('.').collect();
```

Comment:
the big difference here is this must allocate but I am not worried it will be a perf bottleneck.

Reply:
Hmm, yeah, I could use `.next()` twice to avoid the allocation. Either that or open code searching for the first and second periods and taking what falls between, but that's probably not worth it.

Reply:
Isn't this only executed on boot?

Reply:
No, it happens once per request.

Which actually suggests a more important optimization—parse it at config time instead of per request, and cache it.

Reply:
Yes, we should be doing as much as possible up front (obviously). I doubt this will make a difference though.

Reply:
IO takes so much more time that this is fast enough to not even be seen on perf basically.

Reply:
Yeah, the "best" option would be to do it at startup, but the cost here is per-request and should round down to zero relative to doing an HTTP request. If it were per-event it would be a bigger deal, but I wouldn't worry too much about it.

Reply:
Too late, already coded it. :shrug: